device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -174,6 +181,90 @@ def get_data_pretty(self):
         return sffbase.get_data_pretty(self, self.dom_data)
 
 
+class SfpEvent:
+    ''' Listen to insert/remove sfp events '''
+
+    PATH_INT_SYSFS = "{0}/{port_name}/qsfp_isr_flags"
+    PATH_INTMASK_SYSFS = "{0}/{port_name}/qsfp_isr_mask"
+    PATH_PRS_SYSFS = "{0}/{port_name}/qsfp_modprs"
+    PRESENT_EN = 0x01
+
+    def __init__(self, num_port, port_info_path):
+        self.num_sfp = num_port
+        self.port_info_path = port_info_path
+        self.__initialize_interrupts()
+
+    def __initialize_interrupts(self):
+        # Initial Interrupt MASK for QSFP, SFP
+        sfp_info_obj = {}
+
+        for index in range(self.num_sfp):
+            port_num = index + 1
+            port_name = "QSFP{}".format(port_num)
+
+            sfp_info_obj[index] = {}
+            sfp_info_obj[index]['intmask_sysfs'] = self.PATH_INTMASK_SYSFS.format(
+                self.port_info_path,
+                port_name=port_name)
+
+            sfp_info_obj[index]['int_sysfs'] = self.PATH_INT_SYSFS.format(
+                self.port_info_path,
+                port_name=port_name)
+
+            sfp_info_obj[index]['prs_sysfs'] = self.PATH_PRS_SYSFS.format(
+                self.port_info_path,
+                port_name=port_name)
+
+            self._write_file(
+                sfp_info_obj[index]["intmask_sysfs"], hex(self.PRESENT_EN))
+
+        self.sfp_info_obj = sfp_info_obj
+
+    def _write_file(self, file_path, data):
+        try:
+            with open(file_path, 'w') as fd:
+                fd.write(str(data))
+                return True
+        except Exception as e:
+            print "Error: unable to read file: %s" % str(e)
+        return False
+
+    def _read_txt_file(self, file_path):
+        try:
+            with open(file_path, 'r') as fd:
+                data = fd.read()
+                return data.strip()
+        except IOError as e:
+            print "Error: unable to read file: %s" % str(e)
+        return None
+
+    def _is_port_device_present(self, port_idx):
+        prs_path = self.sfp_info_obj[port_idx]["prs_sysfs"]
+        is_present = 1 - int(self._read_txt_file(prs_path))
```

Comment:
this expect the "_read_txt_file" to return digit, while the function could do differently,  we should capture exception here or make the _read_txt_file return 0 or 1.

Reply:
updated